File: ios/RoomPlanModule.m
//
//  RoomPlanManager.m
//  RoomPlanApp
//
//  Created by Water on 2/4/25.
//

#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(RoomPlanModule, NSObject)
RCT_EXTERN_METHOD(startScanning:(RCTPromiseResolveBlock)resolve reject:(RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(stopScanning)
@end

File: ios/RoomPlanModule.swift
import Foundation
import UIKit
import RoomPlan
import React

@objc(RoomPlanModule)
class RoomPlanModule: NSObject {
    @objc func startScanning(_ resolve: @escaping RCTPromiseResolveBlock, reject: @escaping RCTPromiseRejectBlock) {
        DispatchQueue.main.async {
            guard let rootViewController = UIApplication.shared.windows.first?.rootViewController else {
                reject("NO_VIEW_CONTROLLER", "Could not find root view controller", nil)
                return
            }
            
            let captureVC = RoomCaptureViewController()
            captureVC.modalPresentationStyle = .fullScreen
            rootViewController.present(captureVC, animated: true, completion: nil)
        }
    }
    
    @objc func stopScanning() {
        DispatchQueue.main.async {
            if let rootViewController = UIApplication.shared.windows.first?.rootViewController {
                rootViewController.dismiss(animated: true, completion: nil)
            }
        }
    }
}
